django/core/env_helpers.py

```diff
@@ -0,0 +1,50 @@
+import os
+
+from .exceptions import ImproperlyConfigured
+
+
+class NoDefaultValue:
+    pass
+
+
+def from_env(name, default=NoDefaultValue, kind=str):
+    """
+    Get a configuration value from the environment.
+
+    Arguments
+    ---------
+    name : str
+        The name of the environment variable to pull from for this
+        setting.
+    default : any
+        A default value of the return type in case the intended
+        environment variable is not set. If this argument is not passed,
+        the environment variable is considered to be required, and
+        ``ImproperlyConfigured`` may be raised.
+    kind : callable
+        A callable that takes a string and returns a value of the return
+        type.
+
+    Returns
+    -------
+    any
+        A value of the type returned by ``kind``.
+
+    Raises
+    ------
+    ImproperlyConfigured
+        If there is no ``default``, and the environment variable is not
+        set.
+    """
+    try:
+        val = os.environ[name]
+    except KeyError:
+        if default is NoDefaultValue:
+            raise ImproperlyConfigured("Missing environment variable {}.".format(name))
+        val = default
+    val = kind(val)
+    return val
+
+
+def boolable(val):
+    return val in ("True", "true", "T", "t", "1", 1, True)
```

Comment:
This is missing other likely candidates: `y` and `yes`
I'd also argue it should be simplified to be case-insensitive so that `TRUE` == `true` (which requires you actually assert the incoming `val` is actuall stringy enough to have `.lower()`, rather than simply doing an equality match against any of the values)

Reply:
I typically use something like the following:

```python
return str(val).lower() in {'1', 't', 'y', 'on', 'true', 'yes'}
```
This works whether you pass a boolean, integer or string, although we are only expecting a string from the environment anyway.

If we wanted to be strict, we should reject invalid values:

```python
if str(val).lower() in {'1', 't', 'y', 'on', 'true', 'yes'}:
    return True
if str(val).lower() in {'0', 'f', 'n', 'off', 'false', 'no'}:
    return False
raise ValueError('Non-boolean string provided.')
```

Reply:
> I typically use something like the following [....]

Yeah, [me too](https://github.com/kezabelle/python-enviable/blob/58bfd0ec086d7b634f1b0d999a8f1532e101f4ef/enviable.py#L179), which is why I think having those same extra validations/valid values is something to converge on.

> This works whether you pass a boolean, integer or string, although we are only expecting a string from the environment anyway.

well, yes and no. Expecting a string only from the environment, yes (same as eg: form data from a querydict). But the actual contract for any helpers would appear to be a union of `str` and `Any` because the `default` can be passed to the callable and the type there is not guaranteed to be a string.
And yes, I agree that invalid data in any helper should raise an exception lest invalid values get interpreted silently as something other than expected.

Reply:
Can we not make this accept a `bool` type directly, rather than use this special function? Something like:

```
type_map = {
   bool: (“True”, ...)
}
```

Then we can index into that dictionary and not require a “boolable” callable?